src/sync/impl/sync_file.hpp

```diff
@@ -59,19 +61,21 @@ void remove_nonempty_dir(const std::string& path);
 
 class SyncFileManager {
 public:
+    using StringPair = std::tuple<std::string, std::string>;
```

Comment:
Is there a reason you opted for tuple? It'd be clearer if this were a struct with named fields.